api/src/main/java/io/opencensus/trace/Tracer.java

```diff
@@ -151,6 +152,78 @@ public final Scope withSpan(Span span) {
   }
 
   /**
+   * Wraps a {@link Runnable} so that it executes with the {@code Span} as the current {@code Span}
+   * and saves the rest of the current Context.
+   *
+   * <p>This method saves the entire {@link io.grpc.Context} when wraps the {@code Runnable}.
+   *
+   * <p>IMPORTANT: this method does not end the {@code Span}, caller must do that within the {@code
+   * Runnable}, or after the {@code Runnable} is executed.
+   *
+   * <pre><code>
+   * class MyClass {
+   *   private static Tracer tracer = Tracing.getTracer();
+   *   void handleRequest(Executor executor) {
+   *     Span span = new Tracer.spanBuilder("MyRunnableSpan");
+   *     executor.execute(tracer.wrap(span, new Runnable() {
+   *      {MessageEventl @}Override
+   *       public void run() {
+   *         try {
+   *           sendResult();
+   *         } finally {
+   *           span.end();
+   *         }
+   *       }
+   *     }));
+   *   }
+   * }
+   * </code></pre>
+   *
+   * @param span the {@code Span} to be set as current.
+   * @param runnable the {@code Runnable} to wrap in the {@code Span}.
+   * @return the wrapped {@code Runnable}.
+   */
+  public final Runnable wrap(Span span, Runnable runnable) {
+    return CurrentSpanUtils.wrap(span, runnable);
+  }
+
+  /**
+   * Wraps a {@link Callable} so that it executes with the {@code Span} as the current {@code Span}
+   * and saves the rest of the current Context.
+   *
+   * <p>This method saves the entire {@link io.grpc.Context} when wraps the {@code Callable}.
+   *
+   * <p>IMPORTANT: this method does not end the {@code Span}, caller must do that within the {@code
```

Comment:
before saying what it doesn't say what it does. Ex this is used to ensure the current span is readable when invoking callbacks.

Reply:
Done.